### ISSUE TITLE
[mainline] (SotA) strange ai behavior

### ISSUE THREAD

**Issue by OrangePumpking** [wesnoth/wesnoth#8362]
### Game and System Information

- **Version: 1.16.11**
- **Downloaded from: Steam**
- **Build info: not custom**
- **OS: Windows 10 I think?**

### Description of the bug

I have been playing SotA on challenging mode in 1.16.11.

In S21 (Against the World), the mermen faction shows a type of weird ai behavior I have never seen before. They leave their keep, don't try to attack, and just move all the way west and then south to get around the main land mass, all the while keeping a protective circle around their leader. When they reach the castle that used to belong to the saurians and nagas, they do attack my units there. Strategically it makes sense for them not to come onto land, but it seemed very strange behavior to me. Also, some enemy leaders stay back in their castle to recruit (e.g. the knights) while others just charge in along with their units even when no enemy units are remotely within range (e.g. the dwarves and the elves), which seemed a tad inconsistent. None of this is actually problematic, but I figured I would mention it since it did seem pretty weird.

PS: I love the game, thanks so much to the wonderful devs to make this possible! <3

### Steps to reproduce the behavior

I don't think I did anything special to illicit this behavior. I send a couple units up north, expecting the mermen would come on the land to fight me but they only send 2 or 3 and migrated with the rest.

### Expected behavior

In the savefile below, you can see the start of the mermen migration.
[Bugsave9.gz](https://github.com/wesnoth/wesnoth/files/14184689/Bugsave9.gz)

The replay below shows the ai behaviour.
[SotA-Against the World replay 20240205-231447.gz](https://github.com/wesnoth/wesnoth/files/14184697/SotA-Against.the.World.replay.20240205-231447.gz)

**Comment by knyghtmare** [comment]
I would not classify this as a bug. It's rather intended, as the original designer's comments
```txt
 # We want the merfolk to pick up villages, but not to start slithering
            # over land towards the player's keep, which they would normally do.
            # To prevent it, we make them avoid everything except water and
            # villages (and their own castle). This is only active through turn 3
            # because a micro-ai is going to take over on turn 4 and send them
            # down the river.
```

**Comment by OrangePumpking** [comment]
Oh cool, that matches exactly what happened. I guess I just got caught aback. Should I delete this bug report then?

**Comment by Wedge009** [comment]
Yes, if feel your issue/question is resolved it's best to close. I'm assuming you wanted to do that here, but you can always re-open if not.
